### ISSUE TITLE
Feature request: OpenStack images data source

### ISSUE THREAD

**Issue by dvianello** [hashicorp/terraform#9738]
Hello there, 

first, kudos to you guys! 😄  Data sources are really changing our lives for the better!
Being able to search for an ami id via `aws_ami` is terrific when dealing with immutable infrastructures. 

However, we (I) wonder if there's any plan to implement something similar to query OpenStack Glance for image ids labeled with certain tags (again, see how `aws_ami` works). This will help us close the loop also in OpenStack using packer to build images and terraform to spin them up.

### Terraform Version
Terraform v0.7.7

### Affected Resource(s)
- compute_instance_v2

### Terraform Configuration Files
```hcl
data "openstack_image" "base_image" {
  most_recent = true

  filter = {
    name   = "tag-key"
    values = ["Component"]
  }

  filter = {
    name   = "tag-value"
    values = ["base_image"]
  }
}

resource "openstack_compute_instance_v2" "base_instance" {
  name = "base-instance"

  image_id    = "${openstack_image.base_image.id}"
  flavor_name = "some-flavor"
  key_pair    = "some-key"

}
```

### Expected Behavior
Looks up the latest version of the image having the `base_image` tag and uses that to spin up the vm.

### Actual Behavior
Nothing, yet! Hopefully something soon! 😏

**Comment by jtopjian** [comment]
@dvianello This sounds like a great idea. Keep an eye on #8965 for general OpenStack Image support in Terraform. This data source will come shortly after. :)

**Comment by dvianello** [comment]
@jtopjian, cool, thanks!

**Comment by jtopjian** [comment]
@dvianello @vmindru Just a quick status update: I started work on this, but ran into a bug (https://github.com/gophercloud/gophercloud/issues/251). Currently waiting for that to be fixed before I can continue.

**Comment by dvianello** [comment]
Hi @jtopjian, thanks for the update! Let's hope that the gophercloud folks will quickly react to that bug :-)

**Comment by stack72** [comment]
Closed via #12097
